openassessment/fileupload/backends/gcs.py

```diff
@@ -0,0 +1,63 @@
+"""GCS Bucket File Upload Backend."""
+
+import logging
+from google.cloud import storage
+
+from ..exceptions import FileUploadInternalError
+from .base import BaseBackend
+
+log = logging.getLogger("openassessment.fileupload.api")  # pylint: disable=invalid-name
+
+
+class GCSBackend(BaseBackend):
+    """ GCS Bucket File Upload Backend. """
+
+    def get_upload_url(self, key, content_type):
+        """Get a signed URL for uploading a file to GCS"""
+        bucket_name, key_name = self._retrieve_parameters(key)
+        try:
+            blob = get_blob_object(bucket_name, key_name)
+            url = blob.generate_signed_url(
+                version="v4",
+                expiration=self.UPLOAD_URL_TIMEOUT,
+                method="PUT",
+                content_type=content_type,
+            )
+            return url
+        except Exception as ex:
+            log.exception("An internal exception occurred while generating an upload URL.")
```

Comment:
Does this message give enough context to be useful when troubleshooting an error?

Reply:
You'll never know if you're catching the `Exception` class, but could know if you're using more specific classes. In this case we'll need to discover which are the important exceptions that the code could raise, by manually inspecting the code.

Reply:
refactored 👍🏼